Version `3.7.x` incorrectly formats `using private val` definition

## Configuration (required) ##

```
version = 3.7.3
runner.dialect = scala3

style = defaultWithAlign

rewrite {
  rules = [
    SortModifiers
  ]

  sortModifiers.order = [
    "override",
    "implicit",
    "private", "protected",
    "sealed", "abstract", "final",
    "lazy"
  ]
}
```

## Command-line parameters (required) ##

When I run scalafmt via CLI like this: `scalafmt Test.scala`

## Steps

Given code like this:

```scala
//> using scala "3.2.1"
//> using option "-new-syntax"

//> using dep "org.typelevel::cats-effect:3.5.0"

import cats.effect.Concurrent

final class Test[F[_]](using private val F: Concurrent[F]) extends AnyVal:
  def both[A](fa1: F[A], fa2: F[A]): F[(A, A)] = F.both(fa1, fa2)
```

## Problem

Scalafmt reports error during format:

```console
~/Test.scala:8: error: [dialect scala3] val expected but identifier found
final class Test[F[_]](private using val F: Concurrent[F]) extends AnyVal:
```

At the same time adding `using` to `rewrite.sortModifiers.order` results in error

```console
Unknown input 'using'. Expected one of: lazy, inline, open, implicit, final, opaque, transparent, sealed, infix, protected, private, override, abstract
```

## Expectation

I would like the Scalafmt to respect `implicit` modifiers order.

Also, `scala` compiler prohibits `using` between `private` & `val`: 
`final class Test[F[_]](private using val F: Concurrent[F])` results in

```console
[error] ./Test.scala:8:32
[error] `val` or `var` expected
[error] final class Test[F[_]](private using val F: Concurrent[F]) extends AnyVal:
```

## Workaround

I've found that by replacing `using` with `implicit` with problem could be mitigated.

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/SortModifiers.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/SortModifiers.scala
@@ -38,7 +38,7 @@ class SortModifiers(implicit ctx: RewriteCtx) extends RewriteSession {
       case p: Term.Param =>
         val start = p.pos.start
         sortMods(p.mods.filterNot { m =>
-          m.is[Mod.ValParam] || m.is[Mod.VarParam] ||
+          m.is[Mod.ValParam] || m.is[Mod.VarParam] || m.is[Mod.Using] ||
           TreeOps.noExplicitImplicit(start, false)(m)
         })
       case _ =>

--- a/scalafmt-tests/src/test/resources/rewrite/SortModifiersScala3.stat
+++ b/scalafmt-tests/src/test/resources/rewrite/SortModifiersScala3.stat
@@ -60,6 +60,5 @@ rewrite.sortModifiers.order = [ "private" ]
 final class Test[F[_]](using private val F: Concurrent[F]) extends AnyVal:
   def both[A](fa1: F[A], fa2: F[A]): F[(A, A)] = F.both(fa1, fa2)
 >>>
-test does not parse
-final class Test[F[_]](private using val F: Concurrent[F]) extends AnyVal:
-                               ^
+final class Test[F[_]](using private val F: Concurrent[F]) extends AnyVal:
+  def both[A](fa1: F[A], fa2: F[A]): F[(A, A)] = F.both(fa1, fa2)

--- a/scalafmt-tests/src/test/resources/rewrite/SortModifiersScala3.stat
+++ b/scalafmt-tests/src/test/resources/rewrite/SortModifiersScala3.stat
@@ -54,3 +54,12 @@ inline transparent def choose(b: Boolean): A = {
   if (b) { new A }
   else { new B }
 }
+<<< #3546
+rewrite.sortModifiers.order = [ "private" ]
+===
+final class Test[F[_]](using private val F: Concurrent[F]) extends AnyVal:
+  def both[A](fa1: F[A], fa2: F[A]): F[(A, A)] = F.both(fa1, fa2)
+>>>
+test does not parse
+final class Test[F[_]](private using val F: Concurrent[F]) extends AnyVal:
+                               ^

